internal/bminventory/inventory.go

```diff
@@ -3308,3 +3336,90 @@ func (b *bareMetalInventory) getOpenshiftVersionFromOCP(log logrus.FieldLogger)
 	splits := strings.Split(openshiftVersion, ".")
 	return splits[0] + "." + splits[1], nil
 }
+
+func (b *bareMetalInventory) createInstalledOCPHosts(ctx context.Context, cluster *common.Cluster, tx *gorm.DB, log logrus.FieldLogger) error {
+	nodes, err := b.k8sClient.ListNodes()
+	if err != nil {
+		log.WithError(err).Errorf("Failed to list OCP nodes")
+		return err
+	}
+
+	for _, node := range nodes.Items {
+		if !b.isOCPNodeReady(&node) {
+			continue
+		}
+		id := strfmt.UUID(uuid.New().String())
+		url := installer.GetHostURL{ClusterID: *cluster.ID, HostID: id}
+		hostname := node.Name
+		role := b.getOCPNodeRole(&node)
+
+		inventory, err := b.getOCPHostInventory(&node)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to create inventory for host %s, cluster %s", id, *cluster.ID)
+			return err
+		}
+
+		host := models.Host{
+			ID:                &id,
+			Href:              swag.String(url.String()),
+			Kind:              swag.String(models.HostKindAddToExistingClusterOCPHost),
+			ClusterID:         *cluster.ID,
+			CheckedInAt:       strfmt.DateTime(time.Now()),
+			UserName:          auth.UserNameFromContext(ctx),
+			Role:              role,
+			RequestedHostname: hostname,
+			Inventory:         inventory,
+		}
+
+		err = b.hostApi.RegisterInstalledOCPHost(&host, tx)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *bareMetalInventory) isOCPNodeReady(node *v1.Node) bool {
```

Comment:
Maybe all those helper functions should be outside of inventory package? 
This file is long enough. not sure if they should be directly related to `b *bareMetalInventory`

Reply:
what do you think of moving them to the k8sclient package? This way we will also get rid of all the k8s imports

Reply:
it's not related directly to the client but it can be a helper function there maybe some node wrapper?